src/Catel.Core/Catel.Core.Shared/Logging/LogManager.cs

```diff
@@ -562,10 +561,14 @@ internal static ICatelLog GetCatelLogger(Type type, bool alwaysLog = false)
 
                     _loggers.Add(name, log);
                 }
-                else if (!(log is ICatelLog))
+                else
                 {
-                    // Handle the unlikely event where a logger with the same name is initialized before the catel logger.
-                    throw new ArgumentException(string.Format("An element with the same key already exists in the {0} and does not implement {1}.", typeof(LogManager).Name, typeof(ICatelLog).Name));
+                    var catelLog = log as ICatelLog;
```

Comment:
We still have 2 casts now:

1. In the else (which is executed every time except the first time
2. Inside the return (which is also executed all the time)

Safe-casting a single time will always work:

```
var catelLog = log as ICatelLog;
if (catelLog == null)
{
    throw exception...
}

return catelLog;
```